base/bucket_gocb.go

```diff
@@ -147,10 +148,12 @@ func GetCouchbaseBucketGoCB(spec BucketSpec) (bucket *CouchbaseBucketGoCB, err e
 	user, pass, _ := spec.Auth.GetCredentials()
 	nodesMetadata, err := cluster.Manager(user, pass).Internal().GetNodesMetadata()
 	if err != nil {
+		_ = goCBBucket.Close()
```

Comment:
Does it make sense to move the nodesMetadata work prior to opening the bucket, so that we don't have to turn around and close it again on error?

Reply:
Yup, makes sense.